swift/extractor/SwiftExtractor.cpp

```diff
@@ -18,9 +18,7 @@
 
 using namespace codeql;
 
-static void extractFile(const SwiftExtractorConfiguration& config,
-                        swift::CompilerInstance& compiler,
-                        swift::SourceFile& file) {
+static void archiveFile(const SwiftExtractorConfiguration& config, swift::SourceFile* file) {
```

Comment:
nit: while the swift frontend library does use pointers, in our codebase I'd start using pointers only if we expect them to be nullable (I know, we don't do that yet in the dispatcher). I'd keep `swift::SourceFile` a reference (also for consistency with `config` and `compiler`).

Reply:
Just for the record: are you suggesting to use references? 😆 

Ok, will switch back to them.